pism/pism
# Implement till friction angle optimization

Implements a till friction angle optimization scheme. This PR replaces #410, which was based on an old PISM version.

To test:
- create a mask specifying locations where ice thickness is held constant:
```
ncap2 -O -s "where(topg < 0 && thk*(910.0/1028.0) < 0 - topg) thk_bc_mask=1;" in.nc out.nc
```
- Set `-geometry.update.use_basal_melt_rate false` to disable geometry changes due to basal melt.
- Add `no_gl_retreat` to "surface model modifiers", i.e. use `-surface simple,no_gl_retreat` or similar.

See `doc/sphinx/manual/modeling-choices/subglacier/basal-strength.rst` for more details.

**Checklist**

- [x] update [`CHANGES.rst`](CHANGES.rst)
- [ ] ~add regression tests~
